game-core/src/main/java/games/strategy/engine/data/GameMap.java

```diff
@@ -271,80 +272,110 @@ public Territory getTerritory(final String s) {
   }
 
   /**
-   * Returns the shortest route between two territories or null if no route exists.
+   * Returns all neighbors within a certain distance of the starting territory that match the
+   * condition. Does NOT include the original/starting territory in the returned Set.
    *
-   * @param t1 start territory of the route
-   * @param t2 end territory of the route
+   * <p>TODO: update to properly consider movement cost not just distance
+   *
+   * @param territory referring territory
+   * @param movementLeft maximal distance of the neighboring territories
    */
-  public Route getRoute(final Territory t1, final Territory t2) {
-    return getRoute(t1, t2, Matches.territoryIsLandOrWater());
+  public Set<Territory> getNeighborsByMovementCost(
+      final Territory territory, final Unit unit, final BigDecimal movementLeft) {
+    Preconditions.checkArgument(
+        movementLeft.compareTo(BigDecimal.ZERO) >= 0,
+        "MovementLeft must be non-negative: " + movementLeft);
+    if (movementLeft.compareTo(BigDecimal.ZERO) == 0) {
+      return Collections.emptySet();
+    }
+    final Set<Territory> start = getNeighbors(territory);
```

Comment:
Is the set of territories truly the `start`? Should the variable be named `neighbors`?

Reply:
Renamed.